tests/components/here_travel_time/test_config_flow.py

```diff
@@ -0,0 +1,589 @@
+"""Test the HERE Travel Time config flow."""
+from unittest.mock import patch
+
+from herepy import HEREError
+from herepy.routing_api import InvalidCredentialsError
+import pytest
+
+from homeassistant import config_entries, data_entry_flow
+from homeassistant.components.here_travel_time.const import (
+    CONF_ARRIVAL,
+    CONF_ARRIVAL_TIME,
+    CONF_DEPARTURE,
+    CONF_DEPARTURE_TIME,
+    CONF_ROUTE_MODE,
+    CONF_TRAFFIC_MODE,
+    DOMAIN,
+    ROUTE_MODE_FASTEST,
+    TRAFFIC_MODE_ENABLED,
+    TRAVEL_MODE_CAR,
+    TRAVEL_MODE_PUBLIC_TIME_TABLE,
+)
+from homeassistant.components.here_travel_time.sensor import (
+    CONF_DESTINATION_ENTITY_ID,
+    CONF_DESTINATION_LATITUDE,
+    CONF_DESTINATION_LONGITUDE,
+    CONF_ORIGIN_ENTITY_ID,
+    CONF_ORIGIN_LATITUDE,
+    CONF_ORIGIN_LONGITUDE,
+)
+from homeassistant.const import (
+    CONF_API_KEY,
+    CONF_MODE,
+    CONF_NAME,
+    CONF_UNIT_SYSTEM,
+    CONF_UNIT_SYSTEM_IMPERIAL,
+    CONF_UNIT_SYSTEM_METRIC,
+)
+from homeassistant.core import HomeAssistant
+
+from .const import (
+    API_KEY,
+    CAR_DESTINATION_LATITUDE,
+    CAR_DESTINATION_LONGITUDE,
+    CAR_ORIGIN_LATITUDE,
+    CAR_ORIGIN_LONGITUDE,
+)
+
+from tests.common import MockConfigEntry
+
+
+@pytest.fixture(name="user_step_result")
+async def user_step_result_fixture(hass: HomeAssistant) -> data_entry_flow.FlowResult:
+    """Provide the result of a completed user step."""
+    init_result = await hass.config_entries.flow.async_init(
+        DOMAIN, context={"source": config_entries.SOURCE_USER}
+    )
+    user_step_result = await hass.config_entries.flow.async_configure(
+        init_result["flow_id"],
+        {
+            CONF_API_KEY: API_KEY,
+            CONF_MODE: TRAVEL_MODE_CAR,
+            CONF_NAME: "test",
+        },
+    )
+    await hass.async_block_till_done()
+    yield user_step_result
```

Comment:
We only need to yield from the fixture if we're inside a context manager or want to tear down things after the yield. Here we can simply return the result.